desc/compute/data_index.py

```diff
@@ -133,6 +136,10 @@ def _decorator(func):
         "desc.geometry.curve.FourierPlanarCurve",
         "desc.geometry.core.Curve",
     ],
+    "desc.coils.SplineXYZCoil": [
+        "desc.geometry.curve.SplineXYZCurve",
+        "desc.geometry.core.Curve",
+    ],
```

Comment:
Could you add these new parameterizations to `tests/test_compute_funs/test_compute_everything` and an assert statement to that test like
```python
assert things.keys() == data_index.keys(), "Missing a parameterization to test against master."
```